src/control/lib/support/log.go

```diff
@@ -0,0 +1,640 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package support
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/lib/hardware"
+	"github.com/daos-stack/daos/src/control/lib/hardware/hwprov"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/daos-stack/daos/src/control/server/config"
+)
+
+// Folder names to copy logs and configs
+const (
+	dmgSystemLogFolder = "DmgSystemLog"     // Copy the dmg command output for DAOS system
+	daosNodeLogFolder  = "DaosNodeLog"      // Copy the dmg command output specific to the storage.
+	daosAgentCmdInfo   = "DaosAgentCmdInfo" // Copy the daos_agent command output specific to the node.
+	systemInfo         = "SysInfo"          // Copy the system related information
+	serverLogs         = "ServerLogs"       // Copy the server/control and helper logs
+	clientLogs         = "ClientLogs"       // Copy the server/control and helper logs
+	daosConfig         = "ServerConfig"     // Copy the server config
+	customLogs         = "CustomLogs"       // Copy the Custom logs
+)
+
+const DmgListDeviceCmd = "dmg storage query list-devices"
+const DmgDeviceHealthCmd = "dmg storage query device-health"
+
+var DmgCmd = []string{
+	"dmg system get-prop",
+	"dmg system query",
+	"dmg system list-pools",
+	"dmg system leader-query",
+	"dmg system get-attr",
+	"dmg network scan",
+	"dmg storage scan",
+	"dmg storage scan -n",
+	"dmg storage scan -m",
+	"dmg storage query list-pools -v",
+	"dmg storage query usage",
+}
+
+var AgentCmd = []string{
+	"daos_agent version",
+	"daos_agent net-scan",
+	"daos_agent dump-topology",
+}
+
+var SystemCmd = []string{
+	"iperf3 --help",
+	"dmesg",
+	"lspci -D",
+	"top -bcn1 -w512",
+}
+
+var ServerLog = []string{
+	"EngineLog",
+	"ControlLog",
+	"HelperLog",
+}
+
+var DaosServerCmd = []string{
+	"daos_server version",
+	"daos_metrics",
+	"dump-topology",
+}
+
+type ProgressBar struct {
+	Start      int  // start int number
+	Total      int  // end int number
+	Steps      int  // Int number be increased per steps
+	JsonOutput bool // Json option to skip progress bar if it's enabled
+}
+
+type Params struct {
+	Config       string
+	Hostlist     string
+	TargetFolder string
+	CustomLogs   string
+	JsonOutput   bool
+	LogFunction  string
+	LogCmd       string
+}
+
+type copy struct {
+	Cmd     string
+	Options string
+}
+
+// Print the progress bar during log collect command
+func PrintProgress(progBar *ProgressBar) {
```

Comment:
suggestion - could make these member functions of ProgressBar.A function signature like:

`func (p *ProgressBar) Display(out io.Writer)`

It would be easy enough to decide within the function whether to print progress or end status.